user/tests/wiring/power_management/power.cpp

```diff
@@ -621,6 +632,178 @@ test(POWER_15_DisableFeatureFlag) {
     }
 }
 
+test(POWER_16_PrepareDisableChargingFeatureFlag) {
+    if (g_state != STATE12_MAGICK) {
+        skip();
+        return;
+    }
+
+    // Place device into known state and reset before advancing to the next test phase
+    // The charge disable feature will not be set
+    SystemPowerConfiguration conf;
+    conf.feature(SystemPowerFeature::PMIC_DETECTION);
+    System.setPowerConfiguration(conf);
+    g_state = STATE13_MAGICK;
+    notifyAndReset(false);
+    // Reset PMIC to start with a clean slate and disconnect USB so that it doesn't interfere
+    Serial.end();
+    PMIC power(true);
+    power.reset();
+    // Finally reset ourselves
+    System.reset();
+}
+
+test(POWER_17_DisableChargingFeatureFlag) {
+    if (g_state == STATE13_MAGICK) {
+        // Check that charging is not disabled after the last reset
+        auto cfg = System.getPowerConfiguration();
+        assertFalse(cfg.isFeatureSet(SystemPowerFeature::DISABLE_CHARGING));
+        // PMIC charging may be enabled or disabled based on power manager state
+        assertTrue(System.batteryState() != BATTERY_STATE_FAULT);
```

Comment:
Nit: `assertNotEqual()`

Reply:
just have to perform type conversion for it